### PR TITLE
Add juniper_graphql_transport_ws crate for new subscription protocol

### LINKED ISSUE #1022: Add graphql-ws support in light of subscriptions-transport-ws deprecation
This was mentioned in #896 but declined at that point. Since then, however, the winds of change have blown in favor of `graphql-ws`. The [graphql-ws protocol](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md) differs from Apollo's original [subscriptions-transport-ws protocol](https://github.com/apollographql/subscriptions-transport-ws/blob/master/PROTOCOL.md), and Apollo are [soon to archive their protocol and are now recommending use of graphql-ws](https://github.com/apollographql/subscriptions-transport-ws).

There is already some friction, as [graphql-ws-client](https://github.com/obmarg/graphql-ws-client) (and most ws clients coming out) only support the new protocol, while juniper only supports the old.

### PR DESCRIPTION
**Pull request by ccbrown**
This adds a new crate: `juniper_graphql_transport_ws`

This crate implements the new graphql-transport-ws protocol as defined [here](https://github.com/enisdenjo/graphql-ws/blob/fbb763a662802a6a2584b0cbeb9cf1bde38158e0/PROTOCOL.md).

This protocol obsoletes the old graphql-ws protocol. Unfortunately the naming situation is a bit confusing as the old protocol uses the "graphql-ws" WebSocket subprotocol and the new protocol's reference implementation is in a repo named "graphql-ws". But the right way to go to me seems to be to map the WebSocket subprotocols into the crate names.

See https://github.com/graphql-rust/juniper/issues/1022 for discussion.

This also adds a `serve_graphql_transport_ws` function to the `juniper_warp` crate, which is analogous to `serve_graphql_ws`. Ideally we would do some auto-negotiation and use the correct protocol intelligently, but it's a little unclear where this should go or what the cleanest way to implement it would be. It is possible to do this in Warp, but Warp doesn't provide first-class support for it. You can read about this exact use-case in https://github.com/seanmonstar/warp/issues/760. I'm punting on that part for a follow-up PR maybe. Same with other `juniper_actix` changes.

Closes #1022.

### CODE REVIEW

File: juniper_graphql_transport_ws/src/lib.rs
Comment (ccbrown) [inline juniper_graphql_transport_ws/src/lib.rs:1]: This file started out as a copy/paste from `juniper_graphql_ws`. The control flow is mostly identical except the new protocol defines several situations in which the server must close the connection with certain codes. So instead of a stream of messages, the connection is a stream of `Output::Message(msg)` / `Output::Close{code, message}`.

**Comment by ccbrown** [comment]
CI failures are unrelated: https://github.com/graphql-rust/juniper/pull/1159

**Comment by tyranron** [comment]
@ccbrown first of all, thank you for spending time on this ♥️

Since the control flow doesn't differ much, maybe it's better to implement this in the existing `juniper_graphql_ws` crate directly, and control the protocol versions via Cargo features?

**Comment by ccbrown** [comment]
> Since the control flow doesn't differ much, maybe it's better to implement this in the existing juniper_graphql_ws crate directly, and control the protocol versions via Cargo features?

A common use-case is for servers to support both simultaneously and negotiate the correct protocol to use on each connection, as discussed in https://github.com/seanmonstar/warp/issues/760. To support that use-case, we can't force the users into one or the other at compile-time.

I also think the changes, while relatively minor, are pervasive enough that it makes switching between the protocols via conditionals very messy and hard to follow.

**Comment by LegNeato** [comment]
I'm going to merge this as-is, thank you!

**Comment by tyranron** [comment]
> A common use-case is for servers to support both simultaneously and negotiate the correct protocol to use on each connection, as discussed in [seanmonstar/warp#760](https://github.com/seanmonstar/warp/issues/760). To support that use-case, we can't force the users into one or the other at compile-time.
> 
> I also think the changes, while relatively minor, are pervasive enough that it makes switching between the protocols via conditionals very messy and hard to follow.

@ccbrown agree. Thanks!
